src/ScrapeRepos.ts

```diff
@@ -0,0 +1,428 @@
+    const starRecordsData: { starred_at: string }[] = []
+    resArray.forEach(({ data }) => {
+      starRecordsData.push(...data)
+    })
+    for (let i = 0; i < starRecordsData.length; ) {
+      starRecordsMap.set(getDateString(starRecordsData[i].starred_at), i + 1)
+      i += Math.floor(starRecordsData.length / maxRequestAmount) || 1
+    }
+  } else {
+    resArray.forEach(({ data }: { data: StargazerRecord[] }, index) => {
+      if (data.length > 0) {
+        const starRecord = data[0]
+        starRecordsMap.set(
+          getDateString(starRecord.starred_at),
+          DEFAULT_PER_PAGE * (requestPages[index] - 1)
+        )
+      }
+    })
+  }
+
+  const starAmount = await getRepoStargazersCount(repo, token)
+  starRecordsMap.set(getDateString(Date.now()), starAmount)
+
+  const starRecords: StarRecord[] = []
+
+  starRecordsMap.forEach((v, k) => {
+    starRecords.push({
+      date: k,
+      count: v
+    })
+  })
+  return starRecords
+}
+
+/** Function generates an array of numbers within a given range
+ * This is needed to only reconstruct the star history from some specific points in the history
+ * And not from looking at every single star recorded
+ * @param {number} from
+ * @param {number} to
+ * @returns Array<number>: An array of numbers within the specified range.
+ */
+function range(from: number, to: number): number[] {
+  const r: number[] = []
+  for (let i = from; i <= to; i++) {
+    r.push(i)
+  }
+  return r
+}
+
+/** Retuns the timestamp of a given date
+ * @param {Date | number | string} t - Date, timestamp or string
+ * @returns Timestamp of the given date
+ */
+function getTimeStampByDate(t: Date | number | string): number {
+  const d = new Date(t)
+
+  return d.getTime()
+}
+
+/** Formats a dat into a string using the specified format
+ * @param {Date | number | string} t
+ * @param {string} format - The format string for the desired date format.
+ * Default is "yyyy/MM/dd hh:mm:ss".
+ * @returns Formatted string
+ */
+function getDateString(t: Date | number | string, format = 'yyyy/MM/dd hh:mm:ss'): string {
+  const d = new Date(getTimeStampByDate(t))
+
+  const year = d.getFullYear()
+  const month = d.getMonth() + 1
+  const date = d.getDate()
+  const hours = d.getHours()
+  const minutes = d.getMinutes()
+  const seconds = d.getSeconds()
+
+  return format
+    .replace('yyyy', String(year))
+    .replace('MM', String(month))
+    .replace('dd', String(date))
+    .replace('hh', String(hours))
+    .replace('mm', String(minutes))
+    .replace('ss', String(seconds))
+}
+
+/** Main function to test the functionality of the different methods
+ * and how to correctly call them and what the intended workflow is about
+ * @todo Github and OpenAI token still have to be added manually
+ * @param {number} timeMode - 0: daily; 1: weekly; 2: monthly => defines the scope of which repos and developers the methods looks at
+ */
+async function main(timeMode: number) {
```

Comment:
I did `npx tsc` to compile it and then run the complied js file. 
When I only the `getRepoInfo` part run I get a list of `null` values as response: 

When I let the `getRepoInfo` part and `getRepoStarRecords` part run I get a `Bad credentials` error.

When I let the whole main function including the openAI request run I get the error `fetch is not defined`. 
The last error is understandable but I don't know about the first two. 
Please test this yourself.

Reply:
Yes you have to put in your own credentials for it to work. These are mine but they are both already banned and not working anymore because I uploaded them to Github

Reply:
I know that and used my own but it did not work still.
And what is with the issue, that  it only returns null. 
Did you run this function yourself?

Reply:
There's a PR up that will add an `env.sh` that will load development env variables

Reply:
Maxi and myself had a call and could resolve it. I made a mistake with the token plus there was an issue with two of the imports.